File: src/test/java/com/dotnomi/fabricdependencyinjection/java/testclasses/multipleconstructors/MultipleInjectConstructors.java
package com.dotnomi.fabricdependencyinjection.java.testclasses.multipleconstructors;

import com.dotnomi.fabricdependencyinjection.annotation.ModInject;
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped;

@ModScoped
public class MultipleInjectConstructors {
  @ModInject
  public MultipleInjectConstructors(ServiceA serviceA) {}

  @ModInject
  public MultipleInjectConstructors(ServiceA serviceA, ServiceB serviceB) {}
}

File: src/test/java/com/dotnomi/fabricdependencyinjection/kotlin/testclasses/multipleconstructors/MultipleInjectConstructors.kt
package com.dotnomi.fabricdependencyinjection.kotlin.testclasses.multipleconstructors

import com.dotnomi.fabricdependencyinjection.annotation.ModInject
import com.dotnomi.fabricdependencyinjection.annotation.ModScoped

@ModScoped
class MultipleInjectConstructors {
    @ModInject
    constructor(serviceA: ServiceA)

    @ModInject
    constructor(serviceA: ServiceA, serviceB: ServiceB)
}
